### DISCUSSION TITLE
RemainingItemsThresholdReachedCommand and RemainingItemsThreshold

### DISCUSSION THREAD

**Discussion by angelru** [xamarin/Xamarin.Forms#11595] [General] [answered]
Hello, 

I think I don't quite understand `RemainingItemsThresholdReachedCommand `and `RemainingItemsThreshold`, I have the value set to 5 `(RemainingItemsThreshold)`, but `RemainingItemsThresholdReachedCommand `is triggered when `ItemsSource `has 1, shouldn't it be triggered when `ItemsSource `has 5 or more?

**Comment by Kapusch** [comment, marked as answer]
As quoted from the doc :
- **RemainingItemsThreshold**, of type **int**, the threshold of items not yet visible in the list at which the **RemainingItemsThresholdReached** event will be fired.
- **RemainingItemsThresholdReachedCommand**, of type **ICommand**, which is executed when the **RemainingItemsThreshold** is reached.

When threshold is set to 5, that means the command will be triggered when only 5 or less items are remaining in your collection source, while not having yet been displayed.

For every new component you want to implement, you should have a look at [Xamarin Forms Samples](https://github.com/xamarin/xamarin-forms-samples) repository.

And don't forget the documentation : https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/collectionview/populate-data#load-data-incrementally
